File: PmcWh.Android/app/src/main/java/com/samho/pmcwhandroid/network/BarcodeCollectionModels.kt
package com.samho.pmcwhandroid.network

import kotlinx.serialization.Serializable

@Serializable
data class BarcodeListDto(
    val listId: Int,
    val name: String,
    val createdAt: String = "",
    val itemCount: Int = 0,
    val totalScans: Int = 0,
)

@Serializable
data class BarcodeListItemDto(
    val itemId: Int,
    val barcode: String,
    val scanCount: Int,
    val lastScannedAt: String = "",
)

@Serializable
data class CreateBarcodeListRequest(
    val name: String,
    val userId: Int? = null,
)

@Serializable
data class ScanBarcodeRequest(
    val barcode: String,
)

@Serializable
data class ScanBarcodeResult(
    val isDuplicate: Boolean,
    val scanCount: Int,
)

File: PmcWh.Api/Models/BarcodeCollectionModels.cs
namespace PmcWh.Api.Models;

public class BarcodeListDto
{
    public int ListId { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public int ItemCount { get; set; }
    public int TotalScans { get; set; }
}

public class BarcodeListItemDto
{
    public int ItemId { get; set; }
    public string Barcode { get; set; } = string.Empty;
    public int ScanCount { get; set; }
    public DateTime LastScannedAt { get; set; }
}

public class CreateBarcodeListRequest
{
    public string Name { get; set; } = string.Empty;
    public int? UserId { get; set; }
}

public class ScanBarcodeRequest
{
    public string Barcode { get; set; } = string.Empty;
}

public class ScanBarcodeResult
{
    public bool IsDuplicate { get; set; }
    public int ScanCount { get; set; }
}
